chore(Badge): Add story for Badge about adding Icon to a badge

benprosen opened the pull request:
Minor addition to the Badge component:
- There are times where we want to display an Icon in a badge. This PR adds a few examples of exactly how to do this type of Badge into the design system as shown below that can be referenced anytime:

<img width="1014" alt="Screenshot 2024-10-24 at 11 31 12 AM" src="https://github.com/user-attachments/assets/99a084b8-9562-40e4-b9bc-8ae04979024e">

wesleymartin85 commented:
@benprosen can you add a second example that uses `size="captionSmall"` for future chromatic visual tests. Add a third example that uses ` colorScheme="successDarkOnLight"`

benprosen commented:
> @benprosen can you add a second example that uses `size="captionSmall"` for future chromatic visual tests. Add a third example that uses ` colorScheme="successDarkOnLight"`

Updated and added more examples
